pkg/types/builder.go

```diff
@@ -151,9 +151,9 @@ func (g *Builder) AddToBuilder(typeNames *TypeNames, r *resource) (*types.Named,
 		g.validationRules += "\n"
 		sp := sanitizePath(p.path)
 		if p.includeInit {
-			g.validationRules += fmt.Sprintf(`// +kubebuilder:validation:XValidation:rule="!('*' in self.managementPolicies || 'Create' in self.managementPolicies || 'Update' in self.managementPolicies) || has(self.forProvider.%s) || has(self.initProvider.%s)",message="%s is a required parameter"`, sp, sp, p.path)
+			g.validationRules += fmt.Sprintf(`// +kubebuilder:validation:XValidation:rule="!('*' in self.managementPolicies || 'Create' in self.managementPolicies || 'Update' in self.managementPolicies) || has(self.forProvider.%s) || (has(self.initProvider) && has(self.initProvider.%s))",message="spec.forProvider.%s is a required parameter"`, sp, sp, p.path)
 		} else {
-			g.validationRules += fmt.Sprintf(`// +kubebuilder:validation:XValidation:rule="!('*' in self.managementPolicies || 'Create' in self.managementPolicies || 'Update' in self.managementPolicies) || has(self.forProvider.%s)",message="%s is a required parameter"`, sp, p.path)
+			g.validationRules += fmt.Sprintf(`// +kubebuilder:validation:XValidation:rule="!('*' in self.managementPolicies || 'Create' in self.managementPolicies || 'Update' in self.managementPolicies) || has(self.forProvider.%s)",message="spec.forProvider.%s is a required parameter"`, sp, p.path)
```

Comment:
How would it behave with the following?

```yaml
apiVersion: eventhub.azure.upbound.io/v1beta1
kind: EventHub
metadata:
  name: example-eventhub
spec:
  initProvider: 
    partitionCount: 2
    messageRetention: 1
```

Not sure if this is a valid scenario though 😅 But it should still not fail with `Invalid value: "object": no such key: forProvider` I assume.

Reply:
`spec.forProvider` is required, while `spec.initProvider` is not. That is that behaviour was even happeneing. If we remove `spec.forProvider` we get this:
```bash
The EventHub "example-eventhub" is invalid: 
* spec.forProvider: Required value
* <nil>: Invalid value: "null": some validation rules were not checked because the object was invalid; correct the existing errors to complete validation
```

Reply:
BTW I intentionally didnt put 
```
spec.forProvider.%s or spec.initProvider.%s is a required parameter
```
due to `initProvider still being alpha and to avoid confusing people. But I dont feel strongly about that. My worry is that somebody would set a value in `spec.initProvider` while the alpha feature is not turned on. WDYT?

Reply:
This sounds reasonable to me.